onnxruntime/core/session/inference_session.h

```diff
@@ -68,8 +68,13 @@ struct SessionOptions {
   // the prefix of the profile file. The current time will be appended to the file name.
   std::basic_string<ORTCHAR_T> profile_file_prefix = ORT_TSTR("onnxruntime_profile_");
 
-  std::string session_logid;                 ///< logger id to use for session output
-  unsigned session_log_verbosity_level = 0;  ///< applies to session load, initialization, etc
+  std::string session_logid;  ///< logger id to use for session output
+
+  /// Log severity for the inference session. Applies to session load, initialization, etc.
+  /// See https://github.com/microsoft/onnxruntime/blob/master/include/onnxruntime/core/common/logging/severity.h
+  /// Default = -1 (use default logger severity)
+  int session_log_severity_level = -1;
```

Comment:
Why do we need two integers to control the log level? And they have different types.

If we must have two, should we expose all of them through the C API?

Reply:
One is the severity which maps to an enum. The other is the VLOG level if logging severity is verbose (VLOG is disabled otherwise) which is an arbitrary unsigned int. The idea is that as you increase the VLOG level you get more and more verbose output enabled. 

As you shouldn't have to set the severity at the overall, inference and run level, I'm using -1 to indicate the severity should be inherited, so 'int' for the first value to allow that. Seemed simpler to use than unsigned max.

I agree that there are too many places to set these things. Maybe we can get rid of the Run level severity and vlog level and just use the inference session values. Or maybe go one step further and just have an overall severity and vlog level, and the inference session and run loggers use whatever that is set to at the time they're created. We'd be removing fields from SessionOptions and RunOptions though which are part of the public api. 

@pranavsharma what are your thoughts on removing some of these options and simplifying?

Reply:
Once the session is intialized, I should be able to control the run log level at will to debug issues. Using the session values prevents you from this flexibility. Typically the session is created once during (say) the initialization of a process/application and then Run() is called for inference for every incoming request. In my past experience with serving applications, it has been super helpful to turn on verbose logging (with some user defined trace - in this case logid) for a single request through a parameter and then investigate the logs against this trace.

Reply:
So there's a good argument for per-Run severity/vlog level. Do we need the inference session levels or are the overall values fine?

Reply:
I think they're fine. There are only 4 settings - 2 each for session and run and both have clear meanings. If the meanings are not clear, documentation can help. Moreover we can set good defaults to make it easier for users.

Reply:
Probably we should see if we can keep the types consistent.